src/main/resources/templates/new_view_datamart.html

```diff
@@ -825,21 +825,35 @@ <h3 class="panel-title">Selected</h3>
                 "render": function(data, type, row, meta) {
                     var result = '';
                     if ((row[8] && row[8] === "Dimension") || row[3] === "None" || !row[3]) {
-                        if (data && data.length > 0 ) {
+                        if (data && data[0].length > 0 ) {
                             result += '<ul class="va-list" style="list-style:none;padding:0">\n';
-                            for (var i = 0; i < data.length; i++) {
+                            var projectionVMsSize = data[0].length;
+
+                            // First element of data indicates the default mapping value (or null))
+                            var defaultVM = data[1];
+                            for (var i = 0; i < projectionVMsSize; i++) {
                                 result +=
                                     '<li style="margin-top:10px">\n' +
                                     '<div>\n' +
-                                    '<input type="text" class="form-control va-input" pattern="[A-z- ]{1,30}" value="'+ data[i][0] + '" id="val-' + meta.row + '-' + i + '"/>\n' +
+                                    '<input onclick="handleVaInputOnClick(event)" oninput="updateProjectionVMs(event)" type="text" class="form-control va-input" value="'+ data[0][i][0] + '" id="val-' + meta.row + '-' + i + '"/>\n' +
                                     '<i class="glyphicon glyphicon-arrow-right"></i>\n' +
-                                    '<input type="text" class="form-control va-input" pattern="[A-z- ]{1,30}" value="'+ data[i][1] + '" id="ali-' + meta.row + '-' + i + '"/>\n' +
+                                    '<input onclick="handleVaInputOnClick(event)" oninput="updateProjectionVMs(event)" type="text" class="form-control va-input" value="'+ data[0][i][1] + '" id="ali-' + meta.row + '-' + i + '"/>\n' +
                                     '<button type="button" class="btn btn-xs btn-danger" onclick="removeVM(event,' + meta.row + ',' + i +')">\n' +
                                     '<i class="glyphicon glyphicon-remove"></i>\n' +
                                     '</button>\n' +
                                     '</div>\n' +
                                     '</li>';
                             }
+                            // Add default VM input box and checkbox based on the values retrieved above
+                            // OnClick and OnInput events are over-ridden to stop edit-projection pop-up and re-sync VMs
+                            if (defaultVM != null) {
+                                result += '<input onclick="handleVaInputOnClick(event)" oninput="updateProjectionVMs(event)" type="text" value="' + defaultVM + '" class="form-control va-input" id="defaultVM-' + meta.row + '" placeholder="Default value mapping" style="margin:10px; margin-left:0px;">';
```

Comment:
Would this be cleaner if it was like this instead?
string_1 = '<input onclick="handleVaInputOnClick(event)" oninput="updat....{0}'
if (defaultVM != null) {
string_1.format(defaultVM)
}
else {
string_1.format(empty_string)
}

That way, there's less repeated code